src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleTypeSymbol.vb

```diff
@@ -0,0 +1,987 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Collections.Immutable
+Imports System.Globalization
+Imports System.Runtime.InteropServices
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Collections
+Imports Microsoft.CodeAnalysis.RuntimeMembers
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+    Friend Class TupleTypeSymbol
+        Inherits WrappedNamedTypeSymbol
+
+        Private _locations As ImmutableArray(Of Location)
+
+        Private _elementLocations As ImmutableArray(Of Location)
+
+        Private _elementNames As ImmutableArray(Of String)
+
+        Private _elementTypes As ImmutableArray(Of TypeSymbol)
+
+        Private _lazyMembers As ImmutableArray(Of Symbol)
+
+        Private _lazyFields As ImmutableArray(Of FieldSymbol)
+
+        Private _lazyUnderlyingDefinitionToMemberMap As SmallDictionary(Of Symbol, Symbol)
+
+        Friend Const RestPosition As Integer = 8
+
+        Friend Const TupleTypeName As String = "ValueTuple"
+
+        Private Shared tupleTypes As WellKnownType() = New WellKnownType() {WellKnownType.System_ValueTuple_T1, WellKnownType.System_ValueTuple_T2, WellKnownType.System_ValueTuple_T3, WellKnownType.System_ValueTuple_T4, WellKnownType.System_ValueTuple_T5, WellKnownType.System_ValueTuple_T6, WellKnownType.System_ValueTuple_T7, WellKnownType.System_ValueTuple_TRest}
```

Comment:
it does not seem to be a common style, so I keep these as is.

---
In reply to: [71550316](https://github.com/dotnet/roslyn/pull/12596#discussion_r71550316) [](ancestors = 71550316)